### ISSUE TITLE
When my classification number is 60,000, the gpu utilization rate plummets.

### ISSUE THREAD

**Issue by lianyingteng** [tensorpack/tensorpack#1104]
If you're asking about an unexpected problem which you do not know the root cause,
use this template. __PLEASE DO NOT DELETE THIS TEMPLATE, FILL IT__:

### 1. What you did:

First of all, it should be noted that tensorpack uses very high GPU utilization when training the classification model in ImageNet (1000 classification). But when I use tensorpack for my large-scale classification task (65000 classification), GPU utilization will fall off the cliff.

> The command I executed is as follows:
`python train_embedding_softmax.py lmdb --train-lmdb /data1/xinghun/datasets/embeddding_datasets_goodsId/6w_goodsid_pair_train.lmdb --labels-file /data1/xinghun/datasets/embeddding_datasets_goodsId/labels_goods_id_20190310.txt --gpus 0,1 --train-sample-num 2613258 --batch-size 64 --parallel 10 --data-format NCHW --pretain-model /data/xinghun/checkpoint/tensorpack_pretrained_model/ImageNet-ResNet50.npz --depth 50 --mode preact --embedding-num 512`

### 2. What you observed:

（1）The training log is as follows：<br>
> /block2/conv3/bn/gamma, group3/block2/conv3/bn/mean/EMA, group3/block2/conv3/bn/variance/EMA, linear/W, linear/b 
> ^[[32m[0310 17:56:16 @sessinit.py:217] ^[[0m Restoring 185 variables from dict ...
^[[32m[0310 17:56:30 @base.py:242]^[[0m Graph Finalized.
^[[32m[0310 17:56:30 @concurrency.py:38]^[[0m Starting EnqueueThread QueueInput/input_queue ...
^[[32m[0310 17:56:30 @graph.py:73]^[[0m Running Op sync_variables/sync_variables_from_main_tower ...
^[[32m[0310 17:56:32 @param.py:158]^[[0m [HyperParamSetter] At global_step=0, learning_rate is set to 0.050000
^[[32m[0310 17:56:33 @base.py:274]
^[[0m Start Epoch 1 ...
^[[32m[0310 17:56:33 @input_source.py:551]^[[0m Pre-filling StagingArea ...
^[[32m[0310 17:56:33 @input_source.py:555]^[[0m 1 element was put into StagingArea on each tower.
^[[32m[0310 20:42:04 @base.py:284]^[[0m Epoch 1 (global_step 20416) finished, time:2 hours 45 minutes 30 seconds.
^[[32m[0310 20:42:04 @graph.py:73]^[[0m Running Op sync_variables/sync_variables_from_main_tower ...
^[[32m[0310 20:42:08 @saver.py:79]^[[0m Model saved to train_log/embedding-preact-d50-batch128/model-20416.
^[[32m[0310 20:42:08 @misc.py:109]^[[0m Estimated Time Left: 11 days 23 hours 1 minute 48 seconds
^[[32m[0310 20:42:08 @monitor.py:467]^[[0m GPUUtil/0: 95.75
^[[32m[0310 20:42:08 @monitor.py:467]^[[0m GPUUtil/1: 96.738
^[[32m[0310 20:42:08 @monitor.py:467]^[[0m QueueInput/queue_size: 49.42
^[[32m[0310 20:42:08 @monitor.py:467]^[[0m l2_regularize_loss: 1.7093
^[[32m[0310 20:42:08 @monitor.py:467]^[[0m learning_rate: 0.05
^[[32m[0310 20:42:08 @monitor.py:467]^[[0m train-error-top1: 0.62822
^[[32m[0310 20:42:08 @monitor.py:467]^[[0m train-error-top5: 0.41276
^[[32m[0310 20:42:08 @monitor.py:467]^[[0m xentropy-loss: 3.5806
^[[32m[0310 20:42:08 @group.py:48]^[[0m Callbacks took 4.225 sec in total. ModelSaver: 4.16 seconds
^[[32m[0310 20:42:08 @base.py:274]
^[[0m Start Epoch 2 ...
^[[32m[0311 02:18:59 @base.py:284]^[[0m Epoch 2 (global_step 40832) finished, time:5 hours 36 minutes 50 seconds.
^[[32m[0311 02:18:59 @graph.py:73]^[[0m Running Op sync_variables/sync_variables_from_main_tower ...
^[[32m[0311 02:19:03 @saver.py:79]^[[0m Model saved to train_log/embedding-preact-d50-batch128/model-40832.
^[[32m[0311 02:19:03 @misc.py:109]^[[0m Estimated Time Left: 17 days 23 hours 19 minutes 3 seconds
^[[32m[0311 02:19:03 @monitor.py:467]^[[0m GPUUtil/0: 46.845
^[[32m[0311 02:19:03 @monitor.py:467]^[[0m GPUUtil/1: 47.474
^[[32m[0311 02:19:03 @monitor.py:467]^[[0m QueueInput/queue_size: 2.068e-25
^[[32m[0311 02:19:03 @monitor.py:467]^[[0m l2_regularize_loss: 1.9378
^[[32m[0311 02:19:03 @monitor.py:467]^[[0m learning_rate: 0.05
^[[32m[0311 02:19:03 @monitor.py:467]^[[0m train-error-top1: 0.52031
^[[32m[0311 02:19:03 @monitor.py:467]^[[0m train-error-top5: 0.28942
^[[32m[0311 02:19:03 @monitor.py:467]^[[0m xentropy-loss: 2.7425
^[[32m[0311 02:19:03 @group.py:48]^[[0m Callbacks took 3.741 sec in total. ModelSaver: 2.97 seconds
^[[32m[0311 02:19:03 @base.py:274]
^[[0m Start Epoch 3 ...
^[[32m[0311 13:11:23 @base.py:292]
^[[0m Detected Ctrl-C and exiting main loop.
^[[32m[0311 13:11:23 @parallel.py:80]
^[[0m [PrefetchDataZMQ] Socket closed.`

(2) The tensorboard curve is as follows: <br>
![1](https://user-images.githubusercontent.com/26499150/54106717-ff4ed500-4411-11e9-8eab-f43f8ce119fa.jpg)

### 4. Your environment:
  + Python version: 3.6.7.
  + TF version: 1.12.0.
  + Tensorpack version: 0.9.0.1.
  + If you're not using tensorpack under a normal command line shell (e.g.,
    using an IDE or jupyter notebook), please retry under a normal command line shell. 
  + GPU:  Tesla P40 (two)

**Comment by ppwwyyxx** [comment]
As the issue template said:

> If you expect higher speed, please first read http://tensorpack.readthedocs.io/en/latest/tutorial/performance-tuning.html

It can be seen from the log that your data gets slower in the 2nd epoch and you may want to investigate why.

**Comment by lianyingteng** [comment]
This URL is not available

**Comment by ppwwyyxx** [comment]
It should be available now.

**Comment by ppwwyyxx** [comment]
There seems to be some setting issues with the website. please use https://tensorpack.readthedocs.io/tutorial/performance-tuning.html or https://tensorpack.readthedocs.io/en/latest/tutorial/performance-tuning.html: at least one of the two should work.

**Comment by lianyingteng** [comment]
tks

**Comment by lianyingteng** [comment]
I found the problem of batch size. Previously, each gpu batch_size was 64, now it is 32. It can run very well.

**Comment by ppwwyyxx** [comment]
What happened in your log is that the data becomes slower in the 2nd epoch.
It's unlikely that batch size is the issue unless your batch size is different in each epoch. I would recommend you dig into it further.
